gatsby_fp/src/components/fieldComponents/IntellCont.js

```diff
@@ -1,112 +1,160 @@
 import React from "react"
-const shortid = require("shortid")
+import SubHeading from "./intellContComponents/SubHeading.js"
 
 export default function IntellCont({ intellContList }) {
+  const existingContypes = {
+      book: false,
+      bookChapter: false,
+      bookReview: false,
+      case: false,
+      conferenceProceeding: false,
+      instructionalMaterial: false,
+      journalArticle: false,
+      magTradePub: false,
+      newsletter: false,
+      newspaperArticle: false,
+      software: false,
+      technicalReport: false,
+      textbook: false,
+      other: false
+  }
+  intellContList.forEach(element => {
+    switch(element.contype){
+      case "Book":
```

Comment:
I just heard back that we actually need to include all values of `contypeother` in this dictionary. Therefore, you might want to build the whole dictionary dynamically instead of hard-coding the `contype` values even though we know them ahead of time. Basically, it would consist of [everything found in `contype` except "Other", and everything found in `contypeother`].